x-pack/plugins/cases/public/components/case_view/index.tsx

```diff
@@ -513,27 +514,70 @@ export const CaseView = React.memo(
     refreshRef,
     hideSyncAlerts,
   }: CaseViewProps) => {
-    const { data, isLoading, isError, fetchCase, updateCase } = useGetCase(caseId, subCaseId);
-    if (isError) {
-      return <DoesNotExist allCasesNavigation={allCasesNavigation} caseId={caseId} />;
-    }
-    if (isLoading) {
-      return (
-        <MyEuiFlexGroup gutterSize="none" justifyContent="center" alignItems="center">
-          <EuiFlexItem grow={false}>
-            <EuiLoadingSpinner data-test-subj="case-view-loading" size="xl" />
-          </EuiFlexItem>
-        </MyEuiFlexGroup>
-      );
-    }
-    if (onCaseDataSuccess && data) {
-      onCaseDataSuccess(data);
-    }
+    const {
+      data,
+      resolveOutcome,
+      resolveAliasId,
+      isLoading,
+      isError,
+      fetchCase,
+      updateCase,
+    } = useGetCase(caseId, subCaseId);
+    const { spaces: spacesApi, http } = useKibana().services;
 
-    return (
+    useEffect(() => {
+      if (onCaseDataSuccess && data) {
+        onCaseDataSuccess(data);
+      }
+    }, [data, onCaseDataSuccess]);
+
+    useEffect(() => {
+      if (spacesApi && resolveOutcome === 'aliasMatch') {
+        // CAUTION: the path /cases/:detailName is working in both Observability (/app/observability/cases/:detailName) and
+        // Security Solutions (/app/security/cases/:detailName) plugins. This will need to be changed if this component is loaded
+        // under any another path, passing a path builder function by props from every parent plugin.
+        const newPath = http.basePath.prepend(
+          `cases/${resolveAliasId!}${window.location.search}${window.location.hash}`
+        );
+        spacesApi!.ui.redirectLegacyUrl(newPath, i18n.CASE);
```

Comment:
Why the `!`? Is spaces an optional plugin?

Reply:
It is an optional plugin, the docs say so (https://www.elastic.co/guide/en/kibana/master/sharing-saved-objects.html#sharing-saved-objects-step-3). But you are right, this `!` is not really needed here, I already check that it is defined before 👍

Reply:
Yeah, I missed the check before. That was what I wanted to point out :)